terraform/provider/resource_teleport_provision_token.go

```diff
@@ -94,11 +226,12 @@ func resourceProvisionTokenCreate(ctx context.Context, d *schema.ResourceData, m
 	}
 
 	err = c.UpsertToken(ctx, tV2)
+
 	if err != nil {
-		return diagFromErr(describeErr(err, "token"))
+		return diagFromErr(fmt.Errorf("API Error: %v", err))
 	}
 
-	d.SetId(tV2.GetName())
+	d.SetId(index.(string))
```

Comment:
Could convert index to string somewhere at the beginning, handle conversion error and use the result in all cases like this?